sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -387,6 +392,71 @@ public BlobInputStream openInputStream(BlobInputStreamOptions options, Context c
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public SeekableByteChannel openSeekableByteChannelRead(BlobSeekableByteChannelReadOptions options,
+        Context context) {
+        context = context == null ? Context.NONE : context;
+        options = options == null ? new BlobSeekableByteChannelReadOptions() : options;
+        ConsistentReadControl consistentReadControl = options.getConsistentReadControl() == null
+            ? ConsistentReadControl.ETAG : options.getConsistentReadControl();
+        int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
+        long initialPosition = options.getInitialPosition() == null ? 0 : options.getInitialPosition();
+
+        ByteBuffer initialRange = ByteBuffer.allocate(chunkSize);
+        BlobProperties properties;
+        try (ByteBufferBackedOutputStream dstStream = new ByteBufferBackedOutputStream(initialRange)) {
+            BlobDownloadResponse response = this.downloadStreamWithResponse(dstStream,
+                new BlobRange(initialPosition, (long) initialRange.remaining()), null /*downloadRetryOptions*/,
+                options.getRequestConditions(), false, null, context);
+            properties = ModelHelper.buildBlobPropertiesResponse(response).getValue();
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+
+        initialRange.limit(initialRange.position());
+        initialRange.rewind();
+
+        BlobClientBase behaviorClient = this;
+        BlobRequestConditions requestConditions = options.getRequestConditions();
+        switch (consistentReadControl) {
+            case NONE:
+                break;
+            case ETAG:
+                requestConditions = requestConditions != null ? requestConditions : new BlobRequestConditions();
+                // If etag locking but no explicitly specified etag, use the etag from prefetch
+                if (requestConditions.getIfMatch() == null) {
+                    requestConditions.setIfMatch(properties.getETag());
+                }
+                break;
+            case VERSION_ID:
+                if (properties.getVersionId() == null) {
+                    throw LOGGER.logExceptionAsError(
+                        new UnsupportedOperationException("Versioning is not supported on this account."));
+                } else {
+                    // If version locking but no explicitly specified version, use the latest version from prefetch
+                    if (getVersionId() == null) {
+                        behaviorClient = this.getVersionClient(properties.getVersionId());
+                    }
+                }
+                break;
+            default:
+                throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                    "Concurrency control type not supported."));
```

Comment:
nit: Include the type of concurrency control used here to make it clear what it was.